config/config.go

```diff
@@ -367,7 +380,17 @@ func DiscoverFederation() error {
 
 	result, err := httpClient.Do(req)
 	if err != nil {
-		return errors.Wrapf(err, "Failure when doing federation metadata lookup to %s", discoveryUrl)
+		errMsg := errors.Wrapf(err, "Failure when doing federation metadata lookup to %s", discoveryUrl)
+		// If DNS timeout, we want this to be retryable so return a specific error
+		if netErr, ok := err.(net.Error); ok && netErr.Timeout() {
```

Comment:
If we make the `MetadataErr` wrappable, we could have a global var, MetadataTimeoutErr:

```
var MetadataTimeoutErr *MetadataErr := &MetadataErr{err: "Timeout when querying metadata"}
```

And then here, we could have
```
var netErr net.Error
if errors.As(err, netError) {
    err = MetadataTimeoutErr.Wrap(err)
} else {
    NewMetadataError(err, "Error occurred when querying for metadata")
}
```

Then, we can simplify for the retryable test:

```
if errors.Is(err, config.MetadataTimeoutErr) {
   ...
}
```